.github/workflows/build_and_deploy.yml

```diff
@@ -129,7 +132,8 @@ jobs:
           toolchain: stable
           target: ${{ matrix.combinations.rust }}
           default: true
-      - uses: actions/setup-python@v2
+      - uses: Swatinem/rust-cache@v1.4.0
+      - uses: actions/setup-python@v4
         with:
```

Comment:
could you try to add cache: 'pip' here and in the other setup-python instances to try to speed up CI a bit?